src/PJ_bertin1953.c

```diff
@@ -0,0 +1,109 @@
+/*
+  Created by Jacques Bertin in 1953, this projection was the go-to choice
+  of the French cartographic school when they wished to represent phenomena
+  on a global scale.
+  
+  Formula designed by Philippe Rivière, 2017.
+  https://visionscarto.net/bertin-projection-1953
+
+  Port to PROJ by Philippe Rivière, 21 September 2018
+*/
+
+#define PJ_LIB__
+
+#include <errno.h>
+#include <math.h>
+
+#include "proj.h"
+#include "projects.h"
+
+PROJ_HEAD(bertin1953, "Bertin 1953")
+    "\n\tMisc Sph no inv.";
+
+struct pj_opaque {
+    double w;
+    double m, rm;
+    double cosDeltaPhi, sinDeltaPhi, cosDeltaGamma, sinDeltaGamma, deltaLambda;
+};
+
+
+static XY s_forward (LP lp, PJ *P) {
+    XY xy = {0.0,0.0};
+    struct pj_opaque *Q = P->opaque;
+
+    /* Projection constants */
+    double fu = 1.4, k = 12., w = 1.68;
+
+    /* Aliases */
+    double lambda = lp.lam, phi = lp.phi;
+    
+    /* Variable */
+    double d;
+
+    /* Apply rotation */
+    double cosphi, x, y, z, z0;
+	lambda += Q->deltaLambda;
+    cosphi = cos(phi);
+    x = cos(lambda) * cosphi;
+    y = sin(lambda) * cosphi;
+    z = sin(phi);
+    z0 = z * Q->cosDeltaPhi + x * Q->sinDeltaPhi;
+    lambda = atan2(y * Q->cosDeltaGamma - z0 * Q->sinDeltaGamma,
+       x * Q->cosDeltaPhi - z * Q->sinDeltaPhi);
+    z0 = z0 * Q->cosDeltaGamma + y * Q->sinDeltaGamma;
+    phi = asin(z0);
+
+    lambda = adjlon(lambda);
+
+    /* Adjust pre-projection */
+    if (lambda + phi < -fu) {
+      d = (lambda - phi + 1.6) * (lambda + phi + fu) / 8.;
+      lambda += d;
+      phi -= 0.8 * d * sin(phi + M_PI / 2.);
+    }
+
+    /* Project with Hammer (1.68,2) */
```

Comment:
So this is the same Hammer projection as defined in `PJ_hammer.c`, yes? Why not use that directly by instantiating a Hammer specific `PJ` object with `Q->P_hammer = proj_create(P->ctx, '+proj=hammer ...')` in the projection setup and call `pj_fwd(Q->P_hammer, lp)` here?

This avoids duplicate code and potential bugs in Hammer will be fixed both places if necessary.

Reply:
Yes I wanted to do that, but couldn't find an example in the src/

Reply:
Something similar is done in `PJ_deformationc.`:

https://github.com/OSGeo/proj.4/blob/0b32b840d3161c64d8f1f487b153178528c8c124/src/PJ_deformation.c#L270-L272

https://github.com/OSGeo/proj.4/blob/0b32b840d3161c64d8f1f487b153178528c8c124/src/PJ_deformation.c#L89

You'll have to adjust it a bit but the general idea is there. Also notice that you need to create a custom descructor for the `PJ` object. There's an example for that in the same file as well.

Reply:
Feel free to add this! It's my first contribution to Proj.4 and it had been a looong time since I coded anything in C :-)